### ISSUE TITLE
TF.Keras Load Model predict low accuracy when using ROCm

### ISSUE THREAD

**Issue by jszhtian** [ROCmSoftwarePlatform/tensorflow-upstream#1144]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04.1 LTS
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): tensorflow-rocm 2.3.1
- Python version: 3.8
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: ROCm
- GPU model and memory: RX480 8GB

**Describe the current behavior**
I want to test the model accuracy after traning the model on a Computer Cluster.
When I use ROCm version TensorFlow on Linux, It will give me a extreme low accuracy compare with use pure CPU version Tensorflow on Windows.
![image](https://user-images.githubusercontent.com/23244094/95681183-c161f600-0bde-11eb-83eb-4dfc437abb64.png)
**Describe the expected behavior**
Almost the same accuracy compare with pure CPU code.
**Standalone code to reproduce the issue**

[TF2ROCm.zip](https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/files/5361282/TF2ROCm.zip)

**Comment by xuhuisheng** [comment]
Maybe same problem
https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/issues/1106

**Comment by ielenik** [comment]
Duplicate of #1105

**Comment by ppanchad-amd** [comment]
Hi @jszhtian.  Apologies for the lack response. Do you still need assistance with this ticket? If not, please close the ticket. Thanks!
